src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java

```diff
@@ -1444,12 +1473,32 @@ String printCert(boolean isTsCert, String tab, Certificate c,
         }
 
         if (x509Cert != null) {
+            PublicKey key = x509Cert.getPublicKey();
+            String sigalg = x509Cert.getSigAlgName();
 
-            certStr.append("\n").append(tab).append("[");
-
+            // Process the certificate in the signer's cert chain to see if
+            // weak algorithms are used, and provide warnings as needed.
             if (trustedCerts.contains(x509Cert)) {
+                // If the cert is trusted, only check its key size, but not its
+                // signature algorithm. This is because warning should not be
+                // generated for SHA-1 roots which are not an issue.
```

Comment:
In my opinion, the reason we don't checked the signature is that we trust its origin anyway and we don't verify the signature at all (do we?). On the other hand, since its key is used to sign other certs, we need to make sure the key size is big enough so that no one else is able to recover the key and use it to sign other certs.

Reply:
Yes, I would remove the 2nd sentence that starts with "This is ...". There are plenty of references on the Internet which explain this, so no need to add much detail.

Reply:
Removed.